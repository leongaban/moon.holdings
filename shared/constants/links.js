export const NOMICS_LINK = 'https://nomics.com/'

export const COINBASE_LINK = 'https://www.coinbase.com/join/51af71fbe2dfc50f3d000024'

export const BINANCE_LINK = 'https://www.binance.com/?ref=10066165'

export const CASH_APP_LINK = 'https://cash.me/app/PKLJXVD'
