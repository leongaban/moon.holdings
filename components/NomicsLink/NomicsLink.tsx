import React from 'react'

import { LinksContainer, FuturatumSpan } from '../../styles'
import { NOMICS_LINK } from '../../shared/constants/links'

interface IProps {
  displayAboutLink?: boolean
}

const NomicsLink = (props: IProps) => {
  const { displayAboutLink: display } = props
  const FuturatumLink = () => <span> | Built by <a href="https://futuratum.io">Futuratum.</a></span>

  return (
    <LinksContainer>
      <p>
        Powered by <a href={NOMICS_LINK} target="blank">Nomics APIs.</a> 
        { display && <FuturatumSpan><FuturatumLink/></FuturatumSpan> }
      </p>
    </LinksContainer>
  )
}

export default NomicsLink;
