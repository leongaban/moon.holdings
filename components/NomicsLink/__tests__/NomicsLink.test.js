import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import NomicsLink from '../NomicsLink'
import { NOMICS_LINK } from '../../../shared/constants/links'

describe('<NomicsLink /> component', () => {
  describe('rendering', () => {
    const wrapper = shallow(<NomicsLink />);
    it('should render a component matching the snapshot', () => {
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
      expect(wrapper).toHaveLength(1);
    });
  });

  describe('The Nomics API link', () => {
    it('should be correct', () => {
      expect(NOMICS_LINK).toEqual("https://nomics.com/");
    });
  });
});
