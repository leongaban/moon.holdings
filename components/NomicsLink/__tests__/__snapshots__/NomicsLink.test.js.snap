// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NomicsLink /> component rendering should render a component matching the snapshot 1`] = `
<footerLinks__LinksContainer>
  <p>
    Powered by 
    <a
      href="https://nomics.com/"
      target="blank"
    >
      Nomics APIs.
    </a>
  </p>
</footerLinks__LinksContainer>
`;
